action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -561,16 +570,19 @@ func (stateDB *StateDBAdapter) Empty(evmAddr common.Address) bool {
 
 // RevertToSnapshot reverts the state factory to the state at a given snapshot
 func (stateDB *StateDBAdapter) RevertToSnapshot(snapshot int) {
+	ds, ok := stateDB.suicideSnapshot[snapshot]
+	if !ok && stateDB.panicOnDuplicateRevert {
+		log.L().Panic("Failed to revert to snapshot.", zap.Int("snapshot", snapshot))
```

Comment:
according to offline discussion, this won't happen. Thus, hard fork may not be needed. Add a TODO if you insist.

Two more places we need to handle, especially the second one, which was observed before.
https://github.com/iotexproject/iotex-core/pull/4178/files#diff-d4f14fe3911e27971891bedc8bb75529dfdad400cf2beaf785e93e81f1746fc0R581
https://github.com/iotexproject/iotex-core/pull/4178/files#diff-d4f14fe3911e27971891bedc8bb75529dfdad400cf2beaf785e93e81f1746fc0R653

Reply:
1. for safety add hard-fork, can remove after hard-fork, added TODO
Two more places:
1. the first will be taken care of by this change? after hard-fork it will panic, code won't be able to reach here
2. the second is related to `fixSnapshotOrder` flag (which is mostly fixed by trie.db.patch), but I think there are still some cases that error will happen (mostly seen in 13-16m block heights). so we should keep the code as-is, or what is your suggestion here?

Reply:
1. adding TODO is fine
2. if we believe so, using log.Panic instead of log.Error?
3. We need a follow up issue to capture all the cases and fix them